src/alarms/alarm-handling.md

```diff
@@ -0,0 +1,87 @@
+
+# Introduction
+That document specifies alarm handling for thin-edge.
+
+# Data Semantic of Alarms
+Thin-edge treates alarms as **stateful signals**. Instead to usual **cyclic measured values** (as e.g. measurements of physical units like temperature, humidity, pressure, ...) a **stateful signal** is transfered only on state-change. If such a transfered state-change got lost, the cloud assumes the wrong state until a next state-change occurs. Especially in case of alarms that means a once raised alarm would completely unknown to the cloud.
+
+A lost alam-raise could be even more problematic if the raised alarm requires some cloud-site interaction (or manual interaction). Since the lost alarm is not visible on cloud no intercation will be started. And since not state-change happen again, the alarm will never appear at cloud.
+
+**Consequence:**<br/>
+All alarm state-changes need to be transfered reliable from **Application to thin-edge**, and from **thin-edge to cloud**.
+I.E. alarm state-changes must not got lost, or device software must be able to detect loss and react accordingly (e.g. retry to transfer).
+
+More details about loss off data-transfer and proper reaction see...  TODO: add data-flow-image here!
+
+# Information Set per Alarm
+The specified alarm attributes below are inspired by cumulocity data model for alarms, but shall be re-usable for other clouds where possible.
+
+| Name              | Description   |
+| ----------------- |---------------|
+| ref-string        | Device-unique reference string (kind of alarm ID or alarm name), used to reference the once occured alarm again, e.g. in case of updates (e.g. "temperature_sensor_loss") |

```

Comment:
Is a single topic level with `ref-string` sufficient? For example, for a thermostat data, there would be multiple types of alarms like "measurement_loss" alarm, "threshold_cross" alarm, "desired_value_set" alarm etc. If these could be grouped under a thermostat topic as in `thermostat/measurement_loss`, `thermostat/threshold_breach`, `thermostat/value_update` etc, it would enable subscribing to all alarms from the thermostat with `thermostat/+` topic, which is harder with the current singular topic scheme.